doc/Makefile

```diff
@@ -32,14 +32,17 @@ cleanall: clean
 html: deps
 	@echo "Building HTML documentation."
 ifneq ($(OS),WINNT)
-	$(JULIA_EXECUTABLE) $(call cygpath_w,$(SRCDIR)/make.jl) -- deploy
+	$(JULIA_EXECUTABLE) $(call cygpath_w,$(SRCDIR)/make.jl) -- $(ARGS)
 else
 # work around issue #11727, windows output redirection breaking on buildbot
-	$(JULIA_EXECUTABLE) $(call cygpath_w,$(SRCDIR)/make.jl) -- deploy > docbuild.log 2>&1
+	$(JULIA_EXECUTABLE) $(call cygpath_w,$(SRCDIR)/make.jl) -- $(ARGS) > docbuild.log 2>&1
 	@cat docbuild.log
 endif
 	@echo "Build finished. The HTML pages are in _build/html."
 
+deploy: html
+deploy: ARGS := deploy
```

Comment:
Sets the `ARGS` variable, but only for the `deploy` target, and it gets passed on to all the dependencies, hence I can use it in `html`. https://www.gnu.org/software/make/manual/make.html#Target_002dspecific

But I have no idea if this is a good pattern to use or not, just the only thing that came to mind.